### PR TITLE
Introduce two WebGL samples: triangle and suzanne.

### PR DESCRIPTION
**Pull request by prideout**
This commit has no effect on mobile / desktop builds, it only adds new
targets to the `-p webgl` build.

We will eventually expose a proper JavaScript API, but for now these
samples use a (somewhat under-engineered) `filaweb` framework whereby
the WebAssembly module exposes a small number of C entry points: launch,
render, and resize.

Each sample has two source files: a cpp file and an html file. The cpp
file generates js / wasm pair.  The generated js is simply a loader for
the wasm.

The CMake script creates a pristine "public" folder which contains the
minimal set of files needed to serve the web application using a simple
static file server. The public folder like this:

    /suzanne.js         built by em++ from suzanne.cpp
    /suzanne.wasm       built by em++ from suzanne.cpp
    /suzanne.html       copied from REPO/samples/web
    /filaweb.js         copied from REPO/samples/web
    /favicon.png        copied from REPO/samples/web
    /monkey/*.png       copied from REPO/assets/models
    /monkey/*.filamesh  built by filamesh
    /desert/*           built by cmgen

To decode PNG textures, we use the somewhat unusual approach of using
JavaScript to draw them into a hidden 2D canvas, then reading back the
pixels. This allows us to avoid fattening up the wasm file with a PNG
decoder. An alternative idea would be to pass a DOM Image directly into
glTexImage2D, but this would require some #ifdefing and/or JS injection
in Filament's OpenGL backend.

### CODE REVIEW

File: samples/materials/texturedLit.mat [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+}
+
+fragment {
+    void material(inout MaterialInputs material) {
+        prepareMaterial(material);
+        material.baseColor = texture(materialParams_albedo, getUV0());
+        material.roughness = texture(materialParams_roughness, getUV0()).r;
+        material.metallic = texture(materialParams_metallic, getUV0()).r;
+        material.reflectance = materialParams.reflectance;
```
Comment (romainguy) [inline samples/materials/texturedLit.mat]: Tip: setting material properties enables new code paths in the shaders, which is expensive. The fewer properties you set, the better.

File: samples/web/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+function(add_demo NAME)
+    include_directories(${GENERATION_ROOT})
+    add_executable(${NAME} ${NAME}.cpp
+            filamesh.h
+            filamesh.cpp
+            filaweb.h
+            filaweb.cpp
+            public/monkey/mesh.filamesh
+            public/desert/m0_nx.rgbm
```
Comment (romainguy) [inline samples/web/CMakeLists.txt]: Why do you grab only one cubemap face?

File: samples/materials/texturedLit.mat (hunk 1 above)
Comment (prideout) [reply samples/materials/texturedLit.mat]: Nice catch, I optimized this a bit with a follow-up CL.

File: samples/web/CMakeLists.txt (hunk 2 above)
Comment (prideout) [reply samples/web/CMakeLists.txt]: This merely establishes a file-level dependency between the "executable" and the asset-building step.  Agreed that this need to be cleaned up.

**Pull request merged**